### PR TITLE
Add `.modifiers` and `.semantic_hints` to `Data`, `Class` and `Function`

### LINKED ISSUE #30: How to differentiate class variables and instance variables?
Is there a way currently to differentiate the two ?

I was thinking of naming the instance variables `self.xxx`, but maybe you have something else in mind?

Thanks

### PR DESCRIPTION
**Pull request by NiklasRosenstein**
Closes #30

I found the `semantic_hints` concept quite compelling to represent numerous aspects of variables/constants, functions and classes. Docspec ultimately should be able to represent a superset of common concepts in various programming languages (though right now it is still very Python-centric).

### CODE REVIEW

File: docspec/src/docspec/__init__.py [hunk 1]
```diff
@@ -286,6 +345,23 @@ class Class(HasMembers):
   Represents a class definition.
   """
 
+  class Semantic(enum.Enum):
```
Comment (tristanlatr) [inline docspec/src/docspec/__init__.py:348]: Sorry, I'm giving my 2 cents. 

Why did you add this to the `HasMember` object and not the `Class` object? I don't see a module having all those properties.

Comment (NiklasRosenstein) [reply docspec/src/docspec/__init__.py:348]: @tristanlatr I think you are mistaken, the line you are highlighting here is inside the definition of `Class`. 🙂

Comment (NiklasRosenstein) [reply docspec/src/docspec/__init__.py:348]: I can see how, though. The diff is not very friendly in that regard.

image

image

Comment (tristanlatr) [reply docspec/src/docspec/__init__.py:348]: My bad!
